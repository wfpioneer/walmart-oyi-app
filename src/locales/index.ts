import * as RNLocalize from 'react-native-localize';
import I18n from 'i18n-js';
// @ts-ignore
import moment from 'moment';
import 'moment/locale/es';
import en from './en';
import es from './es';
import zh from './zh';

export const setI18nConfig: () => void = () => {
  // fallback if no available language fits
  const fallback = { languageTag: 'en', isRTL: false };

  const { languageTag } = RNLocalize.findBestAvailableLanguage(['en', 'es', 'zh'])
    || fallback;

  // set i18n-js config
  I18n.translations = { en, es, zh };
  I18n.locale = languageTag;

  // the following is to set up moment to work with spanish localization
  moment.locale(languageTag);
};

export const setLanguage = (language: string) => {
  if (language !== 'en' && language !== 'es' && language !== 'zh') {
    throw new Error('Unsupported language');
  }

  I18n.locale = language;
  moment.locale(language);
};

export const strings: (
  tag: string,
  options?: I18n.TranslateOptions | undefined,
) => string = (tag: string, options: I18n.TranslateOptions | undefined = {}) => I18n.t(tag, options);

export const currencies: (
  value: number,
  options?: I18n.ToCurrencyOptions | undefined
) => string = (value: number, options: I18n.ToCurrencyOptions | undefined = {}) => I18n.toCurrency(value, options);

export const numbers: (
  value: number,
  options?: I18n.ToNumberOptions | undefined
) => string = (value: number, options: I18n.ToNumberOptions | undefined = {}) => I18n.toNumber(value, options);
