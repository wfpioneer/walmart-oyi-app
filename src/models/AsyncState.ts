export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
