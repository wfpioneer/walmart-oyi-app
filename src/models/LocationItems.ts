export interface ZoneItem {
  zoneId: number;
  zoneName: string;
  aisleCount: number;
}

export interface AisleItem {
  aisleId: number;
  aisleName: string;
  sectionCount: number;
}

export interface SectionItem {
  sectionId: number;
  sectionName: string;
}

export interface SectionDetailsItem {
  itemNbr: number;
  itemDesc: string;
  price: number;
}

export interface SectionDetailsPallet {
  palletId: number;
  palletCreateTS: string;
  items?: Omit<SectionDetailsItem, 'price'>[];
}

export interface ReserveDetailsPallet{
  id: number;
  palletCreateTS?: string;
  items: SectionDetailsItem[];
  statusCode: number;
}

export interface LocationItem {
  zone: {
    id: number;
    name: string;
  };
  aisle: {
    id: number;
    name: string;
  };
  section: {
    id: number;
    name: string;
  };
  items: { sectionItems: SectionDetailsItem[] };
  pallets: {
    palletData: Omit<SectionDetailsPallet, 'items'>[];
  }
}

export interface PossibleZone {
  name: string;
  description: string;
}

// eslint-disable-next-line no-shadow
export enum CREATE_FLOW {
  CREATE_ZONE = 'CREATE_ZONE',
  CREATE_AISLE = 'CREATE_AISLE',
  CREATE_SECTION = 'CREATE_SECTION',
  NOT_STARTED = 'NOT_STARTED'
}
