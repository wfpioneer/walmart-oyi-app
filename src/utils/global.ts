export const SNACKBAR_TIMEOUT = 3000;
export const SNACKBAR_TIMEOUT_LONG = 5000;
