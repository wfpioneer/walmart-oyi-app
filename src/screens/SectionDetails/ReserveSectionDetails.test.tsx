import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { NavigationContainer, NavigationContext, NavigationProp } from '@react-navigation/native';
import Toast from 'react-native-toast-message';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react-native';
import store from '../../state/index';
import {
  mockLocationDetails,
  mockLocationDetailsEmpty
} from '../../mockData/locationDetails';
import { AsyncState } from '../../models/AsyncState';
import ReserveSectionDetails, {
  ReserveSectionDetailsScreen,
  combineReserveArrays,
  getPalletDetailsApiHook,
  navListenerHook,
  navigateToPalletManagementHook,
  setPerishableCategoriesHook,
  showActivitySpinner
} from './ReserveSectionDetails';
import {
  mockCombinedReserveData,
  mockPalletDetails
} from '../../mockData/getPalletDetails';
import { ReserveDetailsPallet } from '../../models/LocationItems';
import { strings } from '../../locales';
import { SNACKBAR_TIMEOUT, SNACKBAR_TIMEOUT_LONG } from '../../utils/global';

const navigationProp: NavigationProp<any> = {
  addListener: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getId: jest.fn(),
  getState: jest.fn(),
  dispatch: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(() => true),
  removeListener: jest.fn(),
  reset: jest.fn(),
  setOptions: jest.fn(),
  setParams: jest.fn(),
  navigate: jest.fn()
};

describe('Tests Reserve Section Details Screen', () => {
  const defaultAsyncState: AsyncState = {
    isWaiting: false,
    value: null,
    error: null,
    result: null
  };

  describe('Tests rendering Reserve Section details screen data', () => {
    const sectionDetails: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: mockLocationDetails
      }
    };
    const palletDetails: AsyncState = {
      ...defaultAsyncState,
      result: {
        data: {
          pallets: mockPalletDetails
        }
      }
    };
    const actualNav = jest.requireActual('@react-navigation/native');
    const navContextValue = {
      ...actualNav.navigation,
      isFocused: () => true,
      goBack: jest.fn(),
      addListener: jest.fn(() => jest.fn())
    };
    afterEach(() => {
      jest.clearAllMocks();
    });
    it('render screen with redux', () => {
      const component = (
        <Provider store={store}>
          <NavigationContainer>
            <NavigationContext.Provider value={navContextValue}>
              <ReserveSectionDetails />
            </NavigationContext.Provider>
          </NavigationContainer>
        </Provider>
      );
      const { toJSON } = render(component);
      expect(toJSON()).toMatchSnapshot();
    });
    it('Renders Reserve Details Screen with Mock Reserve Items', () => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <ReserveSectionDetailsScreen
          getSectionDetailsApi={sectionDetails}
          getPalletDetailsApi={palletDetails}
          dispatch={jest.fn()}
          navigation={navigationProp}
          trackEventCall={jest.fn()}
          useEffectHook={jest.fn()}
          palletIds={[]}
          getPalletDetailsComplete={false}
          configComplete={false}
          setConfigComplete={jest.fn()}
          setGetPalletDetailsComplete={jest.fn()}
          palletClicked={false}
          setPalletClicked={jest.fn()}
          useCallbackHook={jest.fn()}
          useFocusEffectHook={jest.fn()}
          palletInfo={{ id: '1234' }}
          perishableCategoriesList={[]}
          perishableCategories=""
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('Renders List Empty Component if sectionDetails returns with no data', () => {
      const sectionDetailsEmpty: AsyncState = {
        ...defaultAsyncState,
        result: {
          data: mockLocationDetailsEmpty
        }
      };
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <ReserveSectionDetailsScreen
          getSectionDetailsApi={sectionDetailsEmpty}
          getPalletDetailsApi={palletDetails}
          dispatch={jest.fn()}
          navigation={navigationProp}
          trackEventCall={jest.fn()}
          useEffectHook={jest.fn()}
          palletIds={[]}
          getPalletDetailsComplete={false}
          configComplete={false}
          setConfigComplete={jest.fn()}
          setGetPalletDetailsComplete={jest.fn()}
          palletClicked={false}
          setPalletClicked={jest.fn()}
          useCallbackHook={jest.fn()}
          useFocusEffectHook={jest.fn()}
          palletInfo={{ id: '1234' }}
          perishableCategoriesList={[]}
          perishableCategories=""
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('Renders List Empty Component if palletDetails returns with no data', () => {
      const palletDetailsEmpty: AsyncState = {
        ...defaultAsyncState,
        result: {
          data: {
            pallets: []
          }
        }
      };
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <ReserveSectionDetailsScreen
          getSectionDetailsApi={sectionDetails}
          getPalletDetailsApi={palletDetailsEmpty}
          dispatch={jest.fn()}
          navigation={navigationProp}
          trackEventCall={jest.fn()}
          useEffectHook={jest.fn()}
          palletIds={[]}
          getPalletDetailsComplete={false}
          configComplete={false}
          setConfigComplete={jest.fn()}
          setGetPalletDetailsComplete={jest.fn()}
          palletClicked={false}
          setPalletClicked={jest.fn()}
          useCallbackHook={jest.fn()}
          useFocusEffectHook={jest.fn()}
          palletInfo={{ id: '1234' }}
          perishableCategoriesList={[]}
          perishableCategories=""
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });

  describe('Tests rendering Get Section Details api responses', () => {
    it('Renders Location Details error message', () => {
      const getPalletDetailsError: AsyncState = {
        ...defaultAsyncState,
        error: 'NetWork Error'
      };
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <ReserveSectionDetailsScreen
          getSectionDetailsApi={defaultAsyncState}
          getPalletDetailsApi={getPalletDetailsError}
          dispatch={jest.fn()}
          navigation={navigationProp}
          trackEventCall={jest.fn()}
          useEffectHook={jest.fn()}
          palletIds={[]}
          getPalletDetailsComplete={false}
          configComplete={false}
          setConfigComplete={jest.fn()}
          setGetPalletDetailsComplete={jest.fn()}
          palletClicked={false}
          setPalletClicked={jest.fn()}
          useCallbackHook={jest.fn()}
          useFocusEffectHook={jest.fn()}
          palletInfo={{ id: '1234' }}
          perishableCategoriesList={[]}
          perishableCategories=""
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('Renders activity indicator when waiting for get pallet details api response', () => {
      const getPalletDetailsIsWaiting: AsyncState = {
        ...defaultAsyncState,
        isWaiting: true
      };
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <ReserveSectionDetailsScreen
          getSectionDetailsApi={defaultAsyncState}
          getPalletDetailsApi={getPalletDetailsIsWaiting}
          dispatch={jest.fn()}
          navigation={navigationProp}
          trackEventCall={jest.fn()}
          useEffectHook={jest.fn()}
          palletIds={[]}
          getPalletDetailsComplete={false}
          configComplete={false}
          setConfigComplete={jest.fn()}
          setGetPalletDetailsComplete={jest.fn()}
          palletClicked={false}
          setPalletClicked={jest.fn()}
          useCallbackHook={jest.fn()}
          useFocusEffectHook={jest.fn()}
          palletInfo={{ id: '1234' }}
          perishableCategoriesList={[]}
          perishableCategories=""
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
    it('Renders activity indicator when waiting for get section details api response', () => {
      const getSectionDetailsIsWaiting: AsyncState = {
        ...defaultAsyncState,
        isWaiting: true
      };
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(
        <ReserveSectionDetailsScreen
          getSectionDetailsApi={getSectionDetailsIsWaiting}
          getPalletDetailsApi={defaultAsyncState}
          dispatch={jest.fn()}
          navigation={navigationProp}
          trackEventCall={jest.fn()}
          useEffectHook={jest.fn()}
          palletIds={[]}
          getPalletDetailsComplete={false}
          configComplete={false}
          setConfigComplete={jest.fn()}
          setGetPalletDetailsComplete={jest.fn()}
          palletClicked={false}
          setPalletClicked={jest.fn()}
          useCallbackHook={jest.fn()}
          useFocusEffectHook={jest.fn()}
          palletInfo={{ id: '1234' }}
          perishableCategoriesList={[]}
          perishableCategories=""
        />
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('CombinedReserveArray Function Tests', () => {
    const { palletData } = mockLocationDetails.pallets;
    it('Returns new Reserved Pallet Array with Mapped ids ', () => {
      const reservedArr = combineReserveArrays(mockPalletDetails, palletData);
      expect(reservedArr).toStrictEqual(mockCombinedReserveData);
    });
    it('Returns Empty Array if one prop is undefined ', () => {
      const emptyArray = combineReserveArrays(undefined, palletData);
      expect(emptyArray).toStrictEqual([]);
    });
  });
  describe('Manage pallet externalized function tests', () => {
    const mockDispatch = jest.fn();
    afterEach(() => {
      jest.clearAllMocks();
    });
    const mockSetGetPalletDetailsComplete = jest.fn();
    const mockSetPalletClicked = jest.fn();
    let mockReservePallets: ReserveDetailsPallet[] | undefined = [];

    it('Tests getPalletDetailsApiHook on success', () => {
      const successApi: AsyncState = {
        ...defaultAsyncState,
        result: {
          status: 200,
          data: {
            pallets: [
              {
                id: 1,
                createDate: 'today',
                expirationDate: 'tomorrow',
                items: []
              }
            ]
          }
        }
      };

      getPalletDetailsApiHook(
        successApi,
        mockReservePallets,
        true,
        mockSetPalletClicked,
        mockDispatch,
        mockSetGetPalletDetailsComplete
      );
      expect(mockDispatch).toBeCalledTimes(1);
      expect(Toast.show).toBeCalledTimes(0);
      expect(mockSetPalletClicked).toHaveBeenCalledWith(false);
      expect(mockSetGetPalletDetailsComplete).toHaveBeenCalledWith(true);
    });

    it('Tests getPalletDetailsApiHook with failed pallets and status 207', () => {
      const failApi: AsyncState = {
        ...defaultAsyncState,
        result: {
          status: 207,
          data: {}
        }
      };

      mockReservePallets = [
        {
          statusCode: 204,
          id: 60,
          createDate: 'today',
          expirationDate: 'tomorrow',
          items: [],
          palletCreateTS: '2022-06-05T00:20:00Z',
          palletId: '60'
        }
      ];
      getPalletDetailsApiHook(
        failApi,
        mockReservePallets,
        false,
        mockSetPalletClicked,
        mockDispatch,
        mockSetGetPalletDetailsComplete
      );
      expect(Toast.show).toHaveBeenCalledWith({
        type: 'error',
        text1: strings('LOCATION.GET_FAILED_PALLETS', { amount: 1 }),
        visibilityTime: SNACKBAR_TIMEOUT_LONG,
        position: 'bottom'
      });
    });
    it('Tests getPalletDetailsApiHook with status 204', () => {
      const apiResult: AsyncState = {
        ...defaultAsyncState,
        result: {
          status: 204,
          data: {}
        }
      };

      getPalletDetailsApiHook(
        apiResult,
        mockReservePallets,
        false,
        mockSetPalletClicked,
        mockDispatch,
        mockSetGetPalletDetailsComplete
      );
      expect(Toast.show).toHaveBeenCalledWith({
        type: 'error',
        text1: strings('LOCATION.PALLET_NOT_FOUND'),
        visibilityTime: SNACKBAR_TIMEOUT,
        position: 'bottom'
      });
    });

    it('Tests getPalletDetailsApiHook on fail', () => {
      const failApi: AsyncState = { ...defaultAsyncState, error: {} };
      getPalletDetailsApiHook(
        failApi,
        mockReservePallets,
        false,
        mockSetPalletClicked,
        mockDispatch,
        mockSetGetPalletDetailsComplete
      );
      expect(Toast.show).toHaveBeenCalledWith({
        type: 'error',
        text1: strings('PALLET.PALLET_DETAILS_ERROR'),
        text2: strings('GENERICS.TRY_AGAIN'),
        visibilityTime: SNACKBAR_TIMEOUT,
        position: 'bottom'
      });
    });
    it('Tests navListenerHook', () => {
      navigationProp.addListener = jest
        .fn()
        .mockImplementation((event, callBack) => {
          callBack();
        });

      navListenerHook(navigationProp, mockDispatch);
      expect(navigationProp.addListener).toBeCalledWith(
        'beforeRemove',
        expect.any(Function)
      );
      expect(mockDispatch).toBeCalledTimes(1);
    });
    it('Tests perishableCategories hook', () => {
      const mockMetConfigComplete = jest.fn();
      setPerishableCategoriesHook(
        [49, 20],
        '',
        mockDispatch,
        mockMetConfigComplete
      );
      expect(mockDispatch).not.toBeCalled();
      setPerishableCategoriesHook([], '', mockDispatch, mockMetConfigComplete);
      expect(mockDispatch).toBeCalledTimes(1);
    });
    it('Tests navigateToPalletManagement Hook', () => {
      const mockPalletInfo = { id: '1234' };
      const mockTrackEventCall = jest.fn();
      const mockIsNotFocused = jest.fn(() => false);
      expect(mockIsNotFocused).not.toBeCalled();
      navigateToPalletManagementHook(
        false,
        false,
        mockTrackEventCall,
        mockPalletInfo,
        navigationProp,
        mockSetGetPalletDetailsComplete,
        mockDispatch
      );
      expect(mockDispatch).not.toBeCalled();
      navigateToPalletManagementHook(
        true,
        true,
        mockTrackEventCall,
        mockPalletInfo,
        navigationProp,
        mockSetGetPalletDetailsComplete,
        mockDispatch
      );
      expect(mockDispatch).toBeCalledTimes(1);
      expect(mockTrackEventCall).toBeCalledTimes(1);
      expect(navigationProp.navigate).toHaveBeenCalledTimes(1);
      expect(navigationProp.navigate).toHaveBeenCalledWith('PalletManagement', { screen: 'ManagePallet' });
      expect(mockSetGetPalletDetailsComplete).toBeCalledWith(false);
    });
    it('test showAactivitySpinner functionality', () => {
      let mockGetPalletDetailsWaiting = false;
      let mockGetPalletDetailsComplete = true;
      let mockGetSectionDetailsWaiting = false;
      let result = showActivitySpinner(
        mockGetPalletDetailsWaiting,
        mockGetPalletDetailsComplete,
        mockGetSectionDetailsWaiting
      );
      expect(result).toStrictEqual(true);

      mockGetPalletDetailsWaiting = true;
      mockGetPalletDetailsComplete = false;
      mockGetSectionDetailsWaiting = false;
      result = showActivitySpinner(
        mockGetPalletDetailsWaiting,
        mockGetPalletDetailsComplete,
        mockGetSectionDetailsWaiting
      );
      expect(result).toStrictEqual(true);

      mockGetPalletDetailsWaiting = false;
      mockGetSectionDetailsWaiting = true;
      result = showActivitySpinner(
        mockGetPalletDetailsWaiting,
        mockGetPalletDetailsComplete,
        mockGetSectionDetailsWaiting
      );
      expect(result).toStrictEqual(true);

      mockGetSectionDetailsWaiting = false;
      result = showActivitySpinner(
        mockGetPalletDetailsWaiting,
        mockGetPalletDetailsComplete,
        mockGetSectionDetailsWaiting
      );
      expect(result).toStrictEqual(false);

      mockGetPalletDetailsComplete = true;
      result = showActivitySpinner(
        mockGetPalletDetailsWaiting,
        mockGetPalletDetailsComplete,
        mockGetSectionDetailsWaiting
      );
      expect(result).toStrictEqual(true);
    });
  });
});
