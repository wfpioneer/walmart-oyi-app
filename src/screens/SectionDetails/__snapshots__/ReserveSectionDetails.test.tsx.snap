// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Reserve Section Details Screen Tests rendering Get Section Details api responses Renders Location Details error message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <mockConstructor
    color="#e57373"
    name="alert"
    size={40}
  />
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.LOCATION.LOCATION_API_ERROR" translation]
  </Text>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e57373",
        "borderRadius": 4,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "95%",
      }
    }
  >
    <Text>
      [missing "en.GENERICS.RETRY" translation]
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`Tests Reserve Section Details Screen Tests rendering Get Section Details api responses Renders activity indicator when waiting for get pallet details api response 1`] = `
<ActivityIndicator
  animating={true}
  color="#0071e9"
  hidesWhenStopped={true}
  size="large"
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`Tests Reserve Section Details Screen Tests rendering Get Section Details api responses Renders activity indicator when waiting for get section details api response 1`] = `
<ActivityIndicator
  animating={true}
  color="#0071e9"
  hidesWhenStopped={true}
  size="large"
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`Tests Reserve Section Details Screen Tests rendering Reserve Section details screen data Renders List Empty Component if palletDetails returns with no data 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <ForwardRef
    ListEmptyComponent={
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 28,
          }
        }
      >
        <mockConstructor
          color="#5FA7F2"
          name="information"
          size={40}
        />
        <Text>
          [missing "en.LOCATION.RESERVE_EMPTY" translation]
        </Text>
      </View>
    }
    data={[]}
    keyExtractor={[Function]}
    renderItem={[Function]}
  />
</View>
`;

exports[`Tests Reserve Section Details Screen Tests rendering Reserve Section details screen data Renders List Empty Component if sectionDetails returns with no data 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <ForwardRef
    ListEmptyComponent={
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 28,
          }
        }
      >
        <mockConstructor
          color="#5FA7F2"
          name="information"
          size={40}
        />
        <Text>
          [missing "en.LOCATION.RESERVE_EMPTY" translation]
        </Text>
      </View>
    }
    data={
      [
        {
          "createDate": "2022-01-05T12:04:50.000Z",
          "expirationDate": "12/31/2022",
          "id": 1,
          "items": [
            {
              "itemDesc": "Nature Valley Honey Nut Granola Bars",
              "itemNbr": 500252,
              "locationType": 8,
              "price": 0,
              "upcNbr": "750500252570",
            },
            {
              "itemDesc": "Cabbage",
              "itemNbr": 123456,
              "locationType": 8,
              "price": 0,
              "upcNbr": "123456000789",
            },
          ],
          "palletCreateTS": "2022-01-03T14:22:34.000Z",
          "palletId": "1",
          "statusCode": 200,
        },
        {
          "createDate": "2022-01-05T12:04:50.000Z",
          "expirationDate": "12/31/2022",
          "id": 123,
          "items": [
            {
              "itemDesc": "Nature Valley Granola Bars",
              "itemNbr": 456789,
              "locationType": 8,
              "price": 1298,
              "upcNbr": "456789000123",
            },
          ],
          "palletCreateTS": "2022-01-03T14:22:34.000Z",
          "palletId": "123",
          "statusCode": 200,
        },
        {
          "createDate": "2022-01-05T12:04:50.000Z",
          "expirationDate": "12/31/2022",
          "id": 456,
          "items": [
            {
              "itemDesc": "MM电池5号24粒.",
              "itemNbr": 123789,
              "locationType": 8,
              "price": 29.9,
              "upcNbr": "123789000456",
            },
          ],
          "palletCreateTS": "2022-01-03T14:22:34.000Z",
          "palletId": "456",
          "statusCode": 200,
        },
      ]
    }
    keyExtractor={[Function]}
    renderItem={[Function]}
  />
</View>
`;

exports[`Tests Reserve Section Details Screen Tests rendering Reserve Section details screen data Renders Reserve Details Screen with Mock Reserve Items 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <ForwardRef
    ListEmptyComponent={
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 28,
          }
        }
      >
        <mockConstructor
          color="#5FA7F2"
          name="information"
          size={40}
        />
        <Text>
          [missing "en.LOCATION.RESERVE_EMPTY" translation]
        </Text>
      </View>
    }
    data={
      [
        {
          "createDate": "2022-01-05T12:04:50.000Z",
          "expirationDate": "12/31/2022",
          "id": 1,
          "items": [
            {
              "itemDesc": "Nature Valley Honey Nut Granola Bars",
              "itemNbr": 500252,
              "locationType": 8,
              "price": 0,
              "upcNbr": "750500252570",
            },
            {
              "itemDesc": "Cabbage",
              "itemNbr": 123456,
              "locationType": 8,
              "price": 0,
              "upcNbr": "123456000789",
            },
          ],
          "palletCreateTS": "2021-01-01",
          "palletId": "1",
          "statusCode": 200,
        },
        {
          "createDate": "2022-01-05T12:04:50.000Z",
          "expirationDate": "12/31/2022",
          "id": 123,
          "items": [
            {
              "itemDesc": "Nature Valley Granola Bars",
              "itemNbr": 456789,
              "locationType": 8,
              "price": 1298,
              "upcNbr": "456789000123",
            },
          ],
          "palletCreateTS": "2021-01-10",
          "palletId": "123",
          "statusCode": 200,
        },
        {
          "createDate": "2022-01-05T12:04:50.000Z",
          "expirationDate": "12/31/2022",
          "id": 456,
          "items": [
            {
              "itemDesc": "MM电池5号24粒.",
              "itemNbr": 123789,
              "locationType": 8,
              "price": 29.9,
              "upcNbr": "123789000456",
            },
          ],
          "palletCreateTS": "1999-10-10",
          "palletId": "456",
          "statusCode": 200,
        },
      ]
    }
    keyExtractor={[Function]}
    renderItem={[Function]}
  />
</View>
`;

exports[`Tests Reserve Section Details Screen Tests rendering Reserve Section details screen data render screen with redux 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 28,
          }
        }
      >
        <mockConstructor
          color="#5FA7F2"
          name="information"
          size={40}
        />
        <Text>
          [missing "en.LOCATION.RESERVE_EMPTY" translation]
        </Text>
      </View>
    }
    collapsable={false}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
    waitFor={
      [
        {
          "current": null,
        },
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 28,
          }
        }
      >
        <Text>
          [missing "en.LOCATION.RESERVE_EMPTY" translation]
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
